### ISSUE TITLE
📎 `noNoninteractiveTabindex` no diagnostics for `React.createElement` api

### ISSUE THREAD

**Issue by vasucp1207** [biomejs/biome#529]
### Description

Rule [noNoninteractiveTabindex](https://biomejs.dev/linter/rules/no-noninteractive-tabindex) does not give diagnostics for `React.createElement` api.

```jsx
React.createElement("div", { tabIndex: 0 })
```

**Comment by nissy-dev** [comment]
Not only this rule, most of rules are not considered about legacy APIs older than React v17. I don't think we have bandwith to take care of these things right now, so I closed at this time. 

We will reconsider about this when we receive more feedback.
